prodactivity-frontend/managers/RoutineManager.tsx

```diff
@@ -34,10 +32,10 @@ export default class RoutineManager {
     async getRoutine(routineId: string): Promise<RoutineViewModel | undefined> {
         try {
             const routine = await FetchHelper.get(`${this.baseURL}/byroutineid/`, { routineId });
-            return Object.assign(new RoutineViewModel, routine);
+            return Object.assign(new RoutineViewModel(), routine);
         } catch (error) {
```

Comment:
I know this wasn't added in your PR, but we might not want to catch errors here and let the code calling getRoutine do it itself. Reason being is that depending on the specific error, we might want to handle things differently in different contexts.

Reply:
In the issue I'm working on now I'm setting up a system for this. The way I'm currently setting it up is catching errors within FetchHelper, and returning a `Result` object with a `result` and `message` field.

The result field has the response from the API, and the message field has an error message that may be returned from the API. To keep things simple, if there was another error somewhere with a response I was just going to add it to that message field.

This way we only need to check for the message field to see if there is an error, instead of having to check the message field and catching errors. If you think it would be better a different way I'm willing to discuss it though.

Reply:
I think this is probably fine then